File: MailerLite/Python/mailerlite_email_sender.py
"""
MailerLite Email Sender
This script provides functionality to send emails to MailerLite using the API.

Author: Rasool Vahdati
Date: 2024/10/07
License: MIT License
"""

import requests

class MailerLiteEmailSender:

    def __init__(self, api_url, api_key):
        self.api_url = api_url
        self.api_key = api_key

    def add_subscriber(self, subscriber_email, group_id):
        endpoint = f"{self.api_url}/groups/{group_id}/subscribers"
        headers = {"Content-Type": "application/json", "X-MailerLite-ApiKey": self.api_key}
        payload = {"email": subscriber_email}

        response = requests.post(endpoint, headers=headers, json=payload)
        return response.status_code == 200
    
    @staticmethod
    def validate_email(email):
        return '@' in email and '.' in email

File: MailerLite/.NET/MailerliteEmailSender.cs

// MailWizzEmailSender.cs
// Description: This class handles MailWizz email API integration in .NET (C#).
// It allows for sending email subscriptions to MailWizz via the MailWizz API.
// Author: Rasool Vahdati

using System;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;

namespace MailerLite
{
    public class MailerLiteEmailSender
    {
        private string _apiUrl;
        private string _apiKey;

        public MailerLiteEmailSender(string apiUrl, string apiKey)
        {
            _apiUrl = apiUrl;
            _apiKey = apiKey;
        }

        public async Task<bool> AddSubscriber(string subscriberEmail, string groupId)
        {
            using (var client = new HttpClient())
            {
                client.DefaultRequestHeaders.Add("X-MailerLite-ApiKey", _apiKey);
                var payload = new { email = subscriberEmail };
                var content = new StringContent(JsonConvert.SerializeObject(payload), Encoding.UTF8, "application/json");

                var response = await client.PostAsync($"{_apiUrl}/groups/{groupId}/subscribers", content);
                return response.IsSuccessStatusCode;
            }
        }
    }
}